devito/passes/iet/definitions.py

```diff
@@ -435,6 +437,9 @@ def _map_function_on_high_bw_mem(self, site, obj, storage, devicerm, read_only=F
         """
         mmap = self.lang._map_to(obj)
 
+        if obj._allocator is CUPY_ALLOC:
```

Comment:
this breaks the abstraction in all sort of ways, unfortunately

You can't access the allocator here to steer compilation

what *really* matters at this point is the `_mem_space`of the object: https://github.com/devitocodes/devito/blob/master/devito/types/basic.py#L39

you shouldn't actually end up here, because GPU-allocated functions shuold have a local mem-space, which in turns naturally prevents them ever enter this point